### ISSUE TITLE
fine-tuning BARTpho with  vinai/bartpho-syllabus, error undefined fairseqs_ids_to_tokens and 'unk' when using vinai/bartpho-word

### ISSUE THREAD

**Issue by HuynhKEn** [VinAIResearch/BARTpho#14]
1. Error undefined fairseqs_ids_to_tokens, I'm trying to fine-tune based on vinai/bartpho-syllabus, can you suggest a solution for this issue?

![image](https://github.com/VinAIResearch/BARTpho/assets/40224237/065a1486-8d38-4200-9c85-db55d8457d53)

"I have tried as instructed here: [https://github.com/VinAIResearch/BARTpho/issues/2#issuecomment-1146988402]"

"I have additionally followed the instructions provided in this CSV file format:
![image](https://github.com/VinAIResearch/BARTpho/assets/40224237/358ea6df-c398-4348-ba42-515f29b9e439)

2. When I use vinai/bartpho-word for training, there is no issue with the training process itself, but the prediction results show the appearance of the 'unk' character.
![image](https://github.com/VinAIResearch/BARTpho/assets/40224237/85e167c5-9b7b-411a-929e-1becb4d44fee)

**Comment by datquocnguyen** [comment]
1. Review the CSV data file to see if there is anything wrong with it, e.g. `""Dieu 601...`
2. With 1/3 epoch (6776/20840), it's not surprising that the model would produce some <unk>. Train with more epochs instead of 1.

**Comment by HuynhKEn** [comment]
@datquocnguyen  You mentioned that you converted your data to JSONL format and attempted to train a model for 10 epochs, with a training loss around 1.43. However, the predictions are performing poorly 
![image](https://github.com/VinAIResearch/BARTpho/assets/40224237/fd2d725d-2795-49e2-bc21-0bd9844989e4)
![image](https://github.com/VinAIResearch/BARTpho/assets/40224237/2525894f-1ee9-4c11-8c04-45128fb053d6)
![image](https://github.com/VinAIResearch/BARTpho/assets/40224237/560ac30d-a082-4b37-9e5f-28cecc4a20b1)
Example about a sample test data :
{"text": "Vụ án_mạng xảy ra_chiều nay , tại 1 ngôi nhà ở hẻm 146 đường Nguyễn_Chế_Nghĩa , P. 12 , Q. 8 , TP.HCM. Công_an địa_phương đang tạm giữ nghi can gây án tên là Phong , 23 tuổi . Nạn_nhân là bà Hoa , cô ruột Phong , 59 tuổi . Theo một_số nhân_chứng , 15h chiều , người_dân xung_quanh thấy đông cán_bộ Công_an chạy vào ngôi nhà trong hẻm nên chạy theo . Vào ngôi nhà trong hẻm , người_dân và Công_an phát_hiện bà Hoa gục chết trên vũng máu . Cạnh thi_thể nạn_nhân có 1 chiếc búa dính máu . Nghi can Phong có_mặt tại ngôi nhà đã bị Công_an khống_chế , đưa về trụ_sở làm_việc . Theo thông_tin sơ_bộ từ Công_an , Phong bị nghi_vấn dùng búa sát_hại cô ruột . Sau khi gây án , Phong gọi điện báo cho Công_an để tự_thú . Bà Hoa không có chồng nên sống với gia_đình em ruột . Phong mới tốt_nghiệp ra trường , đi làm về lĩnh_vực xây_dựng cùng cha . Công_an đang điều_tra làm rõ nguyên_nhân vụ án_mạng .", "target": "Người phụ_nữ tử_vong cạnh chiếc búa và nghi can là cháu ruột , sau khi gây án đã gọi điện báo Công_an ."} ============================> Khi dự đoán Theo thông_tin sơ_bộ từ Công_an, Phong bị nghi_vấn dùng búa sát_hại cô ruột. Sau khi gây án, Phong gọi điện báo cho Công_an để tự_thú. <unk> <unk> <unk> <unk> <unk> <unk> <unk> <unk> <unk> <unk> <unk> <unk> <unk> <unk> <unk> <unk> <unk> <unk> <unk> <unk> <unk> <unk> <unk> <unk> <unk> <unk> <unk> <unk> <unk> <unk> <unk> <unk> <unk> <unk> <unk> <unk> <unk> <unk> <unk> <unk> <unk> <unk> <unk> <unk> <unk> <unk> <unk> <unk> <unk> <unk> <unk> <unk> <unk> <unk> <unk> <unk> <unk> <unk> <unk> <unk> <unk> <unk> <unk> <unk> <unk> <unk> <unk> <unk> <unk> <unk> <unk> <unk> <unk> <unk> <unk> <unk> <unk> <unk> <unk> <unk> <unk> <unk> <unk> <unk> <unk> <unk> <unk> <unk> <unk> <unk> <unk> <unk> <unk> <unk> <unk> <unk> <unk> <unk> <unk> <unk> <unk> <unk>

**Comment by datquocnguyen** [comment]
```
python examples/pytorch/summarization/run_summarization.py \
    --model_name_or_path "/lustre/scratch/client/vinai/users/datnq9/sum_exp_bartpho/bartpho-syllable-base" \
    --do_train \
    --do_eval \
    --do_predict \
    --tokenizer_name "/lustre/scratch/client/vinai/users/datnq9/sum_exp_bartpho/bartpho-syllable-base" \
    --num_train_epochs 1 \
    --learning_rate 0.00005 \
    --warmup_steps 1000 \
    --train_file /lustre/scratch/client/vinai/users/datnq9/sum_exp_bartpho/sum_data/train.json \
    --validation_file /lustre/scratch/client/vinai/users/datnq9/sum_exp_bartpho/sum_data/val.json \
    --test_file /lustre/scratch/client/vinai/users/datnq9/sum_exp_bartpho/sum_data/test.json \
    --output_dir /lustre/scratch/client/vinai/users/datnq9/sum_exp_bartpho/sum_data/tst-summarization \
    --overwrite_output_dir \
    --per_device_train_batch_size=16 \
    --per_device_eval_batch_size=16 \
    --predict_with_generate \
    --evaluation_strategy epoch \
    --save_strategy epoch \
    --load_best_model_at_end \
    --text_column document \
    --summary_column summary
```

A script like the one above, using vinai/bartpho-syllable-base, would obtain a Rouge-L score of 37.5 on the VNDS dataset. Alternatively, you can refer to https://github.com/VinAIResearch/BARTpho/issues/2 for reproducing our results.
